catalogs/catalog-jdbc-common/src/main/java/com/datastrato/gravitino/catalog/jdbc/operation/JdbcTableOperations.java

```diff
@@ -415,7 +416,8 @@ protected JdbcColumn getJdbcColumnFromTable(JdbcTable jdbcTable, String colName)
                         "Column %s does not exist in table %s", colName, jdbcTable.name()));
   }
 
-  protected Connection getConnection(String catalog) throws SQLException {
+  @VisibleForTesting
+  public Connection getConnection(String catalog) throws SQLException {
```

Comment:
noticed that getTables using connection.getSchema() to retrieve tables, the total JDBC logic is muddledness ：（
```
  protected ResultSet getTables(Connection connection) throws SQLException {
    final DatabaseMetaData metaData = connection.getMetaData();
    String databaseName = connection.getSchema();
    return metaData.getTables(databaseName, databaseName, null, JdbcConnectorUtils.getTableTypes());
  }
```

Reply:
> noticed that getTables using connection.getSchema() to retrieve tables, the total JDBC logic is muddledness ：（
> 
> ```
>   protected ResultSet getTables(Connection connection) throws SQLException {
>     final DatabaseMetaData metaData = connection.getMetaData();
>     String databaseName = connection.getSchema();
>     return metaData.getTables(databaseName, databaseName, null, JdbcConnectorUtils.getTableTypes());
>   }
> ```

I have removed the line `String databaseName = connection.getSchema();`, please review it again.

Reply:
> Could you verify whether the problem still exists if getConnection using setSchema(xx) not setCatalog(xx)?
> 
> ```java
>     Connection getConnection(String catalog) {
>        Connection connection = dataSource.getConnection();
>        connection.setCatalog(catalog);
>     }
> ```

It's not work to use `setSchema(xx)`;

Reply:
> > Could you verify whether the problem still exists if getConnection using setSchema(xx) not setCatalog(xx)?
> > ```java
> >     Connection getConnection(String catalog) {
> >        Connection connection = dataSource.getConnection();
> >        connection.setCatalog(catalog);
> >     }
> > ```
> 
> It's not work to use `setSchema(xx)`;

when you test do you remove `String databaseName = connection.getSchema();`

Reply:
> > > Could you verify whether the problem still exists if getConnection using setSchema(xx) not setCatalog(xx)?
> > > ```java
> > >     Connection getConnection(String catalog) {
> > >        Connection connection = dataSource.getConnection();
> > >        connection.setCatalog(catalog);
> > >     }
> > > ```
> > 
> > 
> > It's not work to use `setSchema(xx)`;
> 
> when you test do you remove `String databaseName = connection.getSchema();`

No, the method:
```java
  protected ResultSet getTables(Connection connection) throws SQLException {
    final DatabaseMetaData metaData = connection.getMetaData();
    String databaseName = connection.getSchema();
    return metaData.getTables(databaseName, databaseName, null, JdbcConnectorUtils.getTableTypes());
  }
```
The value of the schema name must be passed as a method parameter, otherwise we cannot remove it.